homeassistant/components/light/yeelight.py

```diff
@@ -97,6 +97,17 @@
     EFFECT_TWITTER,
     EFFECT_STOP]
 
+SERVICE_SET_MODE = 'yeelight_set_mode'
+ATTR_MODE = 'mode'
+
+YEELIGHT_SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+SERVICE_SCHEMA_SET_MODE = YEELIGHT_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_MODE): cv.string
```

Comment:
I think it would be cleaner to validate here already that the string is one of the allowed ones - i.e., a key of `yeelight.enums.PowerMode`, if I understand that correctly. Obviously you need to import the yeelight package for that, so then this schema must be defined inside `setup_platform`.

Reply:
Alright. I will care about!